pkgs/top-level/all-packages.nix

```diff
@@ -61,15 +61,15 @@ with pkgs;
       # thing to to create an earlier thing (leading to infinite recursion) and
       # we also would still respect the stage arguments choices for these
       # things.
-      (if stdenvNoCC.hostPlatform.isDarwin || stdenvNoCC.hostPlatform.useLLVM or false
+      (if stdenvNoCC.hostPlatform.cc == "clang"
        then overrideCC stdenvNoCC buildPackages.llvmPackages.clangNoCompilerRt
        else gccCrossLibcStdenv)
```

Comment:
This seems a bit entangled with `compiler-rt`? Might need an assert in here for them matching.